patches/minecraft/net/minecraft/inventory/Container.java.patch

```diff
@@ -33,3 +33,11 @@
          }
  
          p_94525_2_.func_190917_f(p_94525_3_);
+@@ -803,6 +804,7 @@
+             ItemStack itemstack = ItemStack.field_190927_a;
+             IRecipe irecipe = CraftingManager.func_192413_b(p_192389_3_, p_192389_1_);
+ 
++            if(irecipe != null && (!irecipe.hasForcedLock() || entityplayermp.func_192037_E().func_193830_f(irecipe)))
```

Comment:
Is there an `else` block below? I think you should merge these two conditions because they in fact have much in common (both null checks the recipe and calls `entityplayermp.getRecipeBook().containsRecipe(irecipe)`)